docker-compose.yaml

```diff
@@ -1,21 +1,53 @@
-version: "3"
+version: "3.9"
```

Comment:
I am on LEAP 15.3 and probably my docker/compose is older than yours :sweat_smile:
 
```
ERROR: Version in "./docker-compose.yaml" is unsupported. You might be seeing this error because you're using the wrong Compose file version. Either specify a supported version (e.g "2.2" or "3.3") and place your service definitions under the `services` key, or omit the `version` key and place your service definitions at the root of the file to use version 1.
For more on the Compose file format versions, see https://docs.docker.com/compose/compose-file/
```

```
➜  trento git:(docker-compose) docker-compose version
docker-compose version 1.25.1, build a82fef0
docker-py version: 4.2.0
CPython version: 3.6.15
OpenSSL version: OpenSSL 1.1.1d  10 Sep 2019
```

Reply:
I have: 
``` ~  docker-compose version
Docker Compose version 2.4.1```
@stefanotorresi did we really need this bump for any of the other changes you brought? if so, which?

Reply:
Very nice reference :smile: 
However, what's the proposed solution?

With `version: "3.7"` I am able to make a step forward, but I get this. Some permissions on that repo?
```
➜  trento git:(docker-compose) ✗ docker-compose up -d
Pulling runner (ghcr.io/trento-project/trento-runner:1.0.0)...
ERROR: Head "https://ghcr.io/v2/trento-project/trento-runner/manifests/1.0.0": denied: denied
```

Reply:
So with `version 3.7` I can proceed, but when entering the web container like this 
https://github.com/trento-project/web/blob/f4b684b6be82a03b26596feca0721cde46aa759f/docs/development/hacking_trento.md?plain=1#L53
I have this issue
```
➜  tronto git:(docker-compose) ✗ docker-compose run -it -p 4000:4000 web
Run a one-off command on a service.

For example:

    $ docker-compose run web python manage.py shell

By default, linked services will be started, unless they are already
running. If you do not want to start linked services, use
`docker-compose run --no-deps SERVICE COMMAND [ARGS...]`.

Usage:
    run [options] [-v VOLUME...] [-p PORT...] [-e KEY=VAL...] [-l KEY=VALUE...]
        SERVICE [COMMAND] [ARGS...]

Options:
    -d, --detach          Detached mode: Run container in the background, print
                          new container name.
...
```
The following works instead `docker-compose run -p 4000:4000 web`

Maybe again related to different docker-compose version?

Reply:
Hmm, could be. I'm not sure. By the way, your version of docker-compose is 3 years old!! They have completely rewritten it, so I strongly advise to remove the python version and get the newest one from upstream: https://github.com/docker/compose#linux

Reply:
> your version of docker-compose is 3 years old!
>
:man_facepalming: oh man...

Look, I was anyway thinking of upgrading my workstation soon, so this is just a reason more for me :trollface: 

But seriously, I am not sure if `version 3.7` would be a temporary acceptable tradeoff, and I don't want to force to use _old_ stuff just because of my system.

I'll temporarily continue with the current workflow I have and get up to date soon.